pkg/tsdb/influxdb/influxql/converter/converter.go

```diff
@@ -0,0 +1,423 @@
+				rsp.Frames = append(rsp.Frames, newFrames...)
+			} else {
+				// Frame without time column
+				newFrame := handleTimeSeriesFormatWithoutTimeColumn(valueFields, columns, measurement, query)
+				rsp.Frames = append(rsp.Frames, newFrame)
+			}
+		}
+	}
+
+	return rsp
+}
+
+func readTags(iter *jsonitere.Iterator) (map[string]string, error) {
+	tags := make(map[string]string)
+	for l1Field, err := iter.ReadObject(); l1Field != ""; l1Field, err = iter.ReadObject() {
+		if err != nil {
+			return nil, err
+		}
+		value, err := iter.ReadString()
+		if err != nil {
+			return nil, err
+		}
+		tags[l1Field] = value
+	}
+	return tags, nil
+}
+
+func readColumns(iter *jsonitere.Iterator) (columns []string, err error) {
+	for more, err := iter.ReadArray(); more; more, err = iter.ReadArray() {
+		if err != nil {
+			return nil, err
+		}
+
+		l1Field, err := iter.ReadString()
+		if err != nil {
+			return nil, err
+		}
+		columns = append(columns, l1Field)
+	}
+	return columns, nil
+}
+
+func readValues(iter *jsonitere.Iterator, hasTimeColumn bool) (valueFields data.Fields, err error) {
+	if hasTimeColumn {
+		valueFields = append(valueFields, data.NewField("Time", nil, make([]time.Time, 0)))
+	}
+
+	for more, err := iter.ReadArray(); more; more, err = iter.ReadArray() {
+		if err != nil {
+			return nil, err
+		}
+
+		colIdx := 0
+
+		for more2, err := iter.ReadArray(); more2; more2, err = iter.ReadArray() {
+			if err != nil {
+				return nil, err
+			}
+
+			if hasTimeColumn && colIdx == 0 {
+				// Read time
+				var t float64
+				if t, err = iter.ReadFloat64(); err != nil {
+					return nil, err
+				}
+				valueFields[0].Append(time.UnixMilli(int64(t)).UTC())
+
+				colIdx++
+				continue
+			}
+
+			// Read column values
+			next, err := iter.WhatIsNext()
+			if err != nil {
+				return nil, err
+			}
+
+			switch next {
+			case jsoniter.StringValue:
+				s, err := iter.ReadString()
+				if err != nil {
+					return nil, err
+				}
+				valueFields = maybeCreateValueField(valueFields, data.FieldTypeNullableString, colIdx)
+				maybeFixValueFieldType(valueFields, data.FieldTypeNullableString, colIdx)
+				tryToAppendValue(valueFields, &s, colIdx)
+			case jsoniter.NumberValue:
+				n, err := iter.ReadFloat64()
+				if err != nil {
+					return nil, err
+				}
+				valueFields = maybeCreateValueField(valueFields, data.FieldTypeNullableFloat64, colIdx)
+				maybeFixValueFieldType(valueFields, data.FieldTypeNullableFloat64, colIdx)
+				tryToAppendValue(valueFields, &n, colIdx)
+			case jsoniter.BoolValue:
+				b, err := iter.ReadAny()
+				if err != nil {
+					return nil, err
+				}
+				valueFields = maybeCreateValueField(valueFields, data.FieldTypeNullableBool, colIdx)
+				maybeFixValueFieldType(valueFields, data.FieldTypeNullableBool, colIdx)
+				tryToAppendValue(valueFields, util.ToPtr(b.ToBool()), colIdx)
+			case jsoniter.NilValue:
+				_, _ = iter.Read()
```

Comment:
Why not checking err or response here? comment maybe?

Reply:
It is a `null` value already. should we check regardless?